front/pages/w/[wId]/join.tsx

```diff
@@ -0,0 +1,86 @@
+import { Logo } from "@dust-tt/sparkle";
+import { GetServerSideProps, InferGetServerSidePropsType } from "next";
+import { signIn } from "next-auth/react";
+
+import { GoogleSignInButton } from "@app/components/Button";
+import { isWorkspaceAllowedOnDomain } from "@app/lib/api/workspace";
+
+const { URL = "", GA_TRACKING_ID = "" } = process.env;
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const wId = context.query.wId as string;
+  const cId = context.query.cId as string;
+
+  if (!wId || !cId) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const isAllowedOnDomain = await isWorkspaceAllowedOnDomain(wId);
+  if (!isAllowedOnDomain) {
+    return {
+      redirect: {
+        destination: "/",
+        permanent: false,
+      },
+    };
+  }
+
+  return {
+    props: {
+      wId: wId,
+      cId: cId,
+      baseUrl: URL,
+      gaTrackingId: GA_TRACKING_ID,
+    },
+  };
+};
+
+export default function Join({
+  wId,
+  cId,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  return (
+    <>
```

Comment:
I am not sure about how we want this page to look like so for the sake of submitting a PR to validate the behavior without spending too much time on it, I reused part of the html code of our homepage.